### DISCUSSION TITLE
Ctrl+V to paste from system clipboard when in insert mode

### DISCUSSION THREAD

**Discussion by dev-rifaii** [JetBrains/ideavim#999] [Q&A] [answered]
Hello, I'm using IdeaVim on Windows and currently if I'm in normal mode and I press CTRL+V I go to v-block mode which I'd like to keep. However, when I'm insert mode and I press CTRL + V I get an escape character inserted. I'd like to change the behavior of CTRL+V in insert mode to insert what's actually in the system clipboard (not vim clipboard)

![image](https://github.com/user-attachments/assets/7edee0e1-1472-44e6-9eff-56e43df7f83e)

**Comment by citizenmatt** [comment, marked as answer]
You can use normal Vim mappings to get it working. A quick search shows an answer here: https://superuser.com/a/189198

Add the following to your `~/.ideavimrc` file:

```
vmap <C-c> "+yi
vmap <C-x> "+c
vmap <C-v> c<ESC>"+p
imap <C-v> <C-r><C-o>+
```

**Reply by Dylan-B-Johnson** [reply]
For me this just pastes "+".  But handling CNTRL+V by IDE in the settings fixed it for me.

**Reply by tr4wzified** [reply]
Handling CTRL V by the IDE results in visual block mode not working though? Or did you manage to get both working? For me the above answer also unfortunately just pastes a '+'.

**Reply by tr4wzified** [reply]
Oh, managed to fix it. If anyone else is looking for the same behavior as VsVim, paste this into ~/.ideavimrc:
`inoremap <C-v> <Esc>"+pa`
